stl/inc/xstring

```diff
@@ -2794,24 +2849,37 @@ private:
         const size_type _Right_size   = _Right_data._Mysize;
         const _Elem* const _Right_ptr = _Right_data._Myptr();
         auto& _My_data                = _Mypair._Myval2;
-        if (_Right_size < _BUF_SIZE) { // stay small, don't allocate
-            _Traits::copy(_My_data._Bx._Buf, _Right_ptr, _BUF_SIZE);
-            _My_data._Mysize = _Right_size;
-            _My_data._Myres  = _BUF_SIZE - 1;
-            return;
+#ifdef __cpp_lib_constexpr_string
+        if (!_STD is_constant_evaluated()) // SSO disabled in constexpr context
+#endif // __cpp_lib_constexpr_string
+        {

```

Comment:
Same here, should `if constexpr` be the outer condition

Reply:
I believe this was referencing the structure in `_Construct_lv_contents` which actually doesn't have an `if constexpr`.  I instead used the `bool` approach used elsewhere.